journal/src/main/java/io/camunda/zeebe/journal/file/SegmentWriter.java

```diff
@@ -257,6 +258,7 @@ public void truncate(final long index) {
     if (index < segment.index()) {
       buffer.position(descriptorLength);
       invalidateNextEntry(descriptorLength);
+      lastAsqn = firstAsqn - 1;
```

Comment:
:x: This should be moved before `if`. `lastAsqn` must be reset in both cases. It could happen that in `reset`, while iterating over the records, there are no records with valid asqn. Then the lastAsqn is not reset.

Following test fails.
```
@Test
  void shouldResetAsqnOnTruncateToIndexWithinCurrentSegment() {
    final SegmentedJournal journal = openJournal(2);

    // when
    journal.append(1, data);
    final var expectedLastAsqn = journal.append(2, data).asqn();

    // write to second segment
    final var indexToTruncate = journal.append(-1, data).index();
    journal.append(3, data);
    journal.deleteAfter(indexToTruncate);

    // then
    assertThat(journal.getLastSegment().lastAsqn()).isEqualTo(expectedLastAsqn);
  }
```